homeassistant/components/opower/sensor.py

```diff
@@ -0,0 +1,216 @@
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        value_fn=lambda data: data.cost_to_date,
+    ),
+    OpowerEntityDescription(
+        key="elec_forecasted_cost",
+        name="Current bill electric forecasted cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        value_fn=lambda data: data.forecasted_cost,
+    ),
+    OpowerEntityDescription(
+        key="elec_typical_cost",
+        name="Typical monthly electric cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        value_fn=lambda data: data.typical_cost,
+    ),
+)
+GAS_SENSORS: tuple[OpowerEntityDescription, ...] = (
+    OpowerEntityDescription(
+        key="gas_usage_to_date",
+        name="Current bill gas usage to date",
+        device_class=SensorDeviceClass.GAS,
+        native_unit_of_measurement=UnitOfVolume.CENTUM_CUBIC_FEET,
+        suggested_unit_of_measurement=UnitOfVolume.CENTUM_CUBIC_FEET,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        value_fn=lambda data: data.usage_to_date,
+    ),
+    OpowerEntityDescription(
+        key="gas_forecasted_usage",
+        name="Current bill gas forecasted usage",
+        device_class=SensorDeviceClass.GAS,
+        native_unit_of_measurement=UnitOfVolume.CENTUM_CUBIC_FEET,
+        suggested_unit_of_measurement=UnitOfVolume.CENTUM_CUBIC_FEET,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        value_fn=lambda data: data.forecasted_usage,
+    ),
+    OpowerEntityDescription(
+        key="gas_typical_usage",
+        name="Typical monthly gas usage",
+        device_class=SensorDeviceClass.GAS,
+        native_unit_of_measurement=UnitOfVolume.CENTUM_CUBIC_FEET,
+        suggested_unit_of_measurement=UnitOfVolume.CENTUM_CUBIC_FEET,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        value_fn=lambda data: data.typical_usage,
+    ),
+    OpowerEntityDescription(
+        key="gas_cost_to_date",
+        name="Current bill gas cost to date",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        value_fn=lambda data: data.cost_to_date,
+    ),
+    OpowerEntityDescription(
+        key="gas_forecasted_cost",
+        name="Current bill gas forecasted cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
+        value_fn=lambda data: data.forecasted_cost,
+    ),
+    OpowerEntityDescription(
+        key="gas_typical_cost",
+        name="Typical monthly gas cost",
+        device_class=SensorDeviceClass.MONETARY,
+        state_class=SensorStateClass.TOTAL,
+        suggested_display_precision=0,
```

Comment:
Why do you suggest 0 display precision for all of these? I would think you would at least want cost to be 2 decimal places

Reply:
Added comment:
# suggested_display_precision=0 for all sensors since
# Opower provides 0 decimal points for all these.
# (for the statistics in the energy dashboard Opower does provide decimal points)